src/agent.rs

```diff
@@ -123,32 +139,41 @@ impl EmulatorCoreAgentState {
             scope,
             speed: 0,
             executing: false,
+            initialized: false,
         }
     }
 
     pub fn handle_command(&mut self, command: Command) {
         match command {
             Command::SetCore(_architecture) => {
-                todo!() // Implement once we have a RISCV datapath
+                todo!("Implement setting cores.") // Implement once we have a RISCV datapath
             }
             Command::Initialize(initial_pc, mem) => {
                 self.current_datapath.initialize(initial_pc, mem).unwrap();
+                self.initialized = true;
                 self.updates.changed_memory = true;
                 self.updates.changed_registers = true;
+                self.updates.changed_initialized = true;
```

Comment:
I go into this in a little bit more detail down below but I'd rather just directly send the UpdateInitialized message here instead of collecting it in the updates variable to be sent later on.